### DISCUSSION TITLE
Cannot set Shinobu's scanning interval time

### DISCUSSION THREAD

**Discussion by ResRipper** [Difegue/LANraragi#718] [Q&A] [answered]
**LRR Version and OS**
Ubuntu 22.04 LTS + LANraragi v.0.8.7 in docker

I recently deployed LANraragi to AWS EC2 and used S3 + s3fs to store my archives and find out that the Shinobu seems to cause very high S3 API usage. As I'm not going to change my archives frequently, I was trying to set a higher scan interval time for the Shinobu to reduce access frequency, but haven't found any way to do it in the setting or the source code. Is it possible to manually set a scanning interval time for Shinobu, or it just not how the Shinobu work?

**Comment by Difegue** [comment, marked as answer]
The Background Worker doesn't normally operate on intervals but by subscribing to `inotify` filesystem notifications.  
The initial scan might be what's consuming S3 pretty high? Otherwise it might be that AWS doesn't have proper `inotify` support so the worker falls back on basic polling.  

In any case, I'd suggest using the provided [API](https://sugoi.gitbook.io/lanraragi/v/dev/api-documentation/shinobu-api) to stop the Shinobu worker. (There's no GUI for the stop command, since it's not something I want to actively expose to standard users)
